src/Umbraco.Infrastructure/Trees/TreeNode.cs

```diff
@@ -106,20 +105,8 @@ public string IconFilePath
         {
             get
             {
-                if (IconIsClass)
-                    return string.Empty;
-
-                //absolute path with or without tilde
-                if (Icon.StartsWith("~") || Icon.StartsWith("/"))
-                    return Current.IOHelper.ResolveUrl("~" + Icon.TrimStart('~'));
-
-                //legacy icon path
-
-                // TODO: replace this when we have something other than Current.Configs available
-                var backOfficePath = Current.GlobalSettings.GetUmbracoMvcArea(Current.HostingEnvironment);
-
-
-                return string.Format("{0}images/umbraco/{1}", backOfficePath.EnsureEndsWith("/"), Icon);
+                // TODO: Is this ever actually used? If not remove, if so, add setter.
```

Comment:
I think this is okay.. It is most likely not working all over backoffice, and we should use css icons anyway.

Reply:
Coolio, if it's not working I'll gladly add a factory that treenodecontroller can use that has deps on IOHelper etc